specification/sql/resource-manager/Microsoft.Sql/preview/2022-08-01-preview/StartStopManagedInstanceSchedules.json

```diff
@@ -0,0 +1,389 @@
+          },
+          "204": {
+            "description": "The specified Start/Stop schedule does not exist."
+          }
+        },
+        "x-ms-examples": {
+          "Deletes the managed instance's Start/Stop schedule.": {
+            "$ref": "./examples/StartStopManagedInstanceScheduleDelete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ScheduleItem": {
+      "description": "Schedule info describing when the server should be started or stopped.",
+      "required": [
+        "startDay",
+        "startTime",
+        "stopDay",
+        "stopTime"
+      ],
+      "type": "object",
+      "properties": {
+        "startDay": {
+          "description": "Start day.",
+          "enum": [
+            "Sunday",
+            "Monday",
+            "Tuesday",
+            "Wednesday",
+            "Thursday",
+            "Friday",
+            "Saturday"
+          ],
+          "type": "string",
```

Comment:
Has this been reviewed internally within your team ? 

Why not just pass in a datetime object that implicitly includes the start day ? 

Its not really a great idea to have enums for dates , since these do not localize and are not standard. Also this scheme is not very flexible for example if you want to  schedule it to stop 2 thursdays from now then you do cannot do it in this scheme. Remember that even if you do not support these currently  , it might be required 1-2 years later and your api design needs to account for that right now.

Reply:
To emphasize what Marija said. Our scheduler addresses main customer use case, and that is having predefined "working hours" for their SQL Managed Instances that will repeat every week. That is why this scheduled is designed as list of start-stop pairs.

While we might address new scenarios in the future, so far, we don't have any feedback related to more flexibility in the scheduler (with 100+ customers using it) and from this point, we won't invest into new scenarios in the near future. On the other hand, there are ways how new scenarios can be added in the existing API, for example, introducing "type" property and new set of schedule parameters.

Current approach has already been reviewed and is in use through private preview, and since recently in limited public preview. So this PR request should just convert the "private/hidden" API to the publicly available.